### PR TITLE
Report kustomization error is subscription status

### PR DESCRIPTION
**Pull request by philipwu08**
Story: https://issues.redhat.com/browse/ACM-8446

If kustomization failed:
- on the managed cluster, update:
    - appsubstatus.statuses.subscription.phase = "failed", reason with the error msg
    - subscription.status.phase = "failed
- on the hub, update the cluster and application report, set the result to "failed" 

* [x] I have taken backward compatibility into consideration.

### CODE REVIEW

**Comment by philipwu08** [comment]
/assign @xiangjingli 

Outstanding work: write unit tests
Outstanding tests: multiple managed clusters, local-cluster, standalone, citi git repo

File: addon/manifests/chart/templates/apps.open-cluster-management.io_subscriptionstatuses_crd_v1alpha1.yaml [hunk 1]
```diff
@@ -58,6 +58,14 @@ spec:
                 - lastUpdateTime
                 type: object
               type: array
+            subscription:
+              properties:
+                message:
+                  type: string
+                phase:
+                  description: PackagePhase defines the phasing of a Package
```
Comment (xiangjingli) [inline addon/manifests/chart/templates/apps.open-cluster-management.io_subscriptionstatuses_crd_v1alpha1.yaml:66]: description: PackagePhase defines the phasing of a Package
===>
 description: OverallPhase defines the overall phasing of a subscription

**Comment by xiangjingli** [comment]
And as discussed, The subscription controller watching the subscription phase change is not necessary, I think we can remove the watch 

https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/016724850646930243cb612fcb8f91178ab88793/pkg/controller/subscription/subscription_controller.go#L146-L150

**Comment by philipwu08** [comment]
> And as discussed, The subscription controller watching the subscription phase change is not necessary, I think we can remove the watch
> 
> https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/016724850646930243cb612fcb8f91178ab88793/pkg/controller/subscription/subscription_controller.go#L146-L150

@xiangjingli OK, my misunderstanding here. I've actually move from utils.SubscriptionPredicateFunctions the checking for changes to "phase". You think it's ok to remove watch of subscriptions from the subscription_controller?

File: addon/manifests/chart/templates/apps.open-cluster-management.io_subscriptionstatuses_crd_v1alpha1.yaml (hunk 1 above)
Comment (philipwu08) [reply addon/manifests/chart/templates/apps.open-cluster-management.io_subscriptionstatuses_crd_v1alpha1.yaml:66]: The doc comes from the type:
`// PackagePhase defines the phasing of a Package
type PackagePhase string`

I'll define a new type called SubscriptionPhase, ok?

Comment (xiangjingli) [reply addon/manifests/chart/templates/apps.open-cluster-management.io_subscriptionstatuses_crd_v1alpha1.yaml:66]: ok

**Comment by xiangjingli** [comment]
> You think it's ok to remove watch of subscriptions from the subscription_controller

Yeah,  updating the status always triggers a new reconcile . So watching status change doesn't make sense. But If removing the watch with specific predicate function causes more unnecessary reconcile, we probably have to remain it :-)

**Pull request merged**